products/BellHybrid/services/db/agents/ShuffleQuoteModel.cpp

```diff
@@ -112,12 +121,24 @@ namespace Quotes
         }
 
         const auto currentTimestamp = std::time(nullptr);
-        if (hasCrossedMidnight(currentTimestamp, lastTimestamp)) {
-            settings.setValue(settings::Quotes::randomQuoteIDUpdateTime,
-                              utils::to_string(currentTimestamp),
-                              settings::SettingsScope::Global);
-            return true;
+
+        if (utils::is_number(interval)) {
+            if (isTimeElapsed(currentTimestamp, lastTimestamp, std::chrono::minutes(utils::toNumeric(interval)))) {
```

Comment:
AFAIU if you apply my previous suggestion it should be able to directly pass `utils::toNumeric(interval)` here and avoid the division by `utils::time::secondsInMinute` (if `interval` is expressed in seconds)

Reply:
The interval will be in minutes because the algorithm is written so that the quote duration is checked when the time is updated, i.e. every 1 minute.

Reply:
Is it needed to use `std::chrono::minutes` then? It should be possible to pass the value of `utils::toNumeric(interval)` directly

Reply:
You are right. Done.